Mathlib/Data/ENat/Basic.lean

```diff
@@ -408,6 +408,19 @@ theorem map_natCast_nonneg [AddMonoidWithOne α] [PartialOrder α]
     [AddLeftMono α] [ZeroLEOneClass α] : 0 ≤ n.map (Nat.cast : ℕ → α) := by
   cases n <;> simp
 
+@[simp]
+lemma map_natCast_eq_zero :
+  WithTop.map (Nat.cast : ℕ → ℤ) n = 0 ↔ n = 0 := by
```

Comment:
According to `strictMono_map_iff` just below, it's enough for `Nat.cast` to be strictly monotone. This is true by `Nat.cast_lt`. So can you prove the following?
* `ENat.map (Nat.cast : ℕ → α)` is strictly monotone if `[AddMonoidWithOne α] [PartialOrder α] [AddLeftMono α] [ZeroLEOneClass α] [CharZero α]`
* `ENat.map (Nat.cast : ℕ → α)` is injective if `[AddMonoidWithOne α] [PartialOrder α] [AddLeftMono α] [ZeroLEOneClass α] [CharZero α]`
* `ENat.map (Nat.cast : ℕ → α) n = 0 ↔ n = 0` if `[AddMonoidWithOne α] [PartialOrder α] [AddLeftMono α] [ZeroLEOneClass α] [CharZero α]`

Reply:
I am unsure I understand. Can you please expand your explanation?

Reply:
I have pushed a commit performing what I was requesting

Reply:
Thanks.